cli/cmd/uninstall.go

```diff
@@ -34,16 +34,23 @@ This command provides all Kubernetes namespace-scoped and cluster-scoped resourc
 			}
 
 			if !force {
-				podsList, err := k8sAPI.CoreV1().Pods("").List(cmd.Context(), metav1.ListOptions{})
+				podList, err := k8sAPI.CoreV1().Pods("").List(cmd.Context(), metav1.ListOptions{LabelSelector: k8s.ControllerNSLabel})
 				if err != nil {
 					return err
 				}
-				for _, pod := range podsList.Items {
-					// Skip core control-plane namespace
-					if pod.Namespace != controlPlaneNamespace && healthcheck.ContainsProxy(pod) {
-						return fmt.Errorf("Please uninject proxy containers before uninstalling the control-plane")
+
+				var injectedPods []string
+				for _, pod := range podList.Items {
+					// skip core control-plane namespace
+					if pod.Namespace != controlPlaneNamespace {
+						injectedPods = append(injectedPods, fmt.Sprintf("* %s", pod.Name))
```

Comment:
Could be nice including namespace here? Something like:

```
Please uninject the following pods before uninstalling the control-plane:
        * linkerd-tap-7df756764b-lqzkr
```

Turns into:

```
Please uninject the following pods before uninstalling the control-plane:
        * linkerd-viz/linkerd-tap-7df756764b-lqzkr
```

I realize we don't do this in `check --proxy` output either though so definitely not a blocker.

Reply:
Yep, Good suggestion. Will raise a separate PR fixing this everywhere.